app/src/main/java/com/smashingboxes/epa_prototype_android/MainActivity.java

```diff
@@ -84,41 +91,86 @@ protected void onCreate(Bundle savedInstanceState) {
         detailsAdapter = new JsonDetailsAdapter(this);
         recyclerView.setAdapter(detailsAdapter);
 
+        if(!startSelectUserLocationActivity()) {
+            fetchData();
+        }
+    }
+
+    @Override
+    protected void onRestart() {
+        super.onRestart();
+        fetchData();
+    }
+
+    private void fetchData(){
         getUserProfile();
         getUserActivity();
         getCurrentlySelectedTimeSeries();
     }
 
-    private void getUserProfile(){
+    private boolean startSelectUserLocationActivity() {
+        if (AppStateManager.getInstance(this).getPlace() == null) {
+            try {
+                PlacePicker.IntentBuilder builder = new PlacePicker.IntentBuilder();
+                startActivityForResult(builder.build(this), PICK_PLACE_REQEST);
+                return true;
+            } catch (GooglePlayServicesRepairableException | GooglePlayServicesNotAvailableException e) {
+                e.printStackTrace();
```

Comment:
Not that this would really ever happen, but if it hits the GooglePlayServicesRepairableException exception, it's recommended to display a dialog or something of that nature to prompt the user to download/update play services. `GooglePlayServicesUtil.getErrorDialog(resultCode, context, 9001).show();`